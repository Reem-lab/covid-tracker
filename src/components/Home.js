import { NavLink } from 'react-router-dom';
import { RiVirusLine } from 'react-icons/ri';
import { BsArrowRightCircle } from 'react-icons/bs';
import { VscSearch } from 'react-icons/vsc';
import { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import SearchBar from './SearchBar';
import { displayCountries } from '../redux/actions/AllCountries';
import '../styles/Home.css';

const Home = () => {
  const [location, setLocation] = useState('');
  const countries = useSelector((state) => state.CountriesReducer);
  const filtered = countries.filter((item) => item.country.toLowerCase()
    .includes(location.toLowerCase()));

  const dispatch = useDispatch();

  useEffect(() => {
    if (countries.length === 0) {
      dispatch(displayCountries());
    }
  }, []);

  return (
    <div className="country-section">
      <SearchBar />
      <div className="search">
        <h2 className="h2Search">
          Search by Country
          <VscSearch className="search-icon" />
        </h2>
        <input
          className="input-search"
          placeholder="Search"
          value={location}
          onChange={(e) => { setLocation(e.target.value); }}
        />
      </div>
      <div className="all-countries">
        { !filtered.length ? (
          <div className="heading">No Counrties Information Found ❗❕</div>
        ) : (
          filtered.map((country1) => (
            <div key={country1.country} className="country-card">
              <NavLink to={`/Details/${country1.country}`}>
                <div className="icons">
                  <RiVirusLine className="virsus-icon" />
                  <BsArrowRightCircle className="arrow" />
                </div>
                <div className="Cname">
                  <h2>{country1.country}</h2>
                  <span className="number">{country1.cases}</span>
                </div>
              </NavLink>
            </div>
          )))}
      </div>
    </div>
  );
};

export default Home;
