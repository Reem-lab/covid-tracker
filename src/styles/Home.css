.all-countries {
  display: -ms-grid;
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-template-rows: repeat(6, 1fr);
  gap: 5px;
  border: 5px solid #fff;
}

.country-card {
  background-color: #35548b;
  color: #fff;
}

.country-card:nth-child(2n) {
  background-color: #5485e1;
}

.icons {
  display: flex;
  align-items: flex-start;
  gap: 160px;
  margin: 5px;
}

.virsus-icon {
  width: 70px;
  height: 60px;
}

.arrow {
  width: 20px;
  height: 20px;
}

.Cname {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  margin: 7px;
}
