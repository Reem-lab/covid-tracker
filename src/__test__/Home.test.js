/* eslint-disable no-unused-vars */
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import Home from '../components/Home';
import store from '../redux/store';

describe('Test the Countries list', () => {
  it('renders correctly', () => {
    const rocket = [
      {
        country: 'Afghanstan',
        cases: 222,
      },
    ];

    const { container } = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    const message = screen.getByText('No Counrties Information Found ❗❕');
    expect(message).toMatchSnapshot();
  });
});
