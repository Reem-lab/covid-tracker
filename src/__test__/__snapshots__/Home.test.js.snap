// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Countries list renders correctly 1`] = `
<div
  class="heading"
>
  No Counties Information Found ❗❕
</div>
`;
